moveit_planners/trajopt/trajopt/src/collision_terms.cpp

```diff
@@ -0,0 +1,633 @@
+    CalcCollisions(x, dist_results);
+    m_cache.put(key, dist_results);
+  }
+}
+
+SingleTimestepCollisionEvaluator::SingleTimestepCollisionEvaluator(
+                planning_scene::PlanningSceneConstPtr& planning_scene, std::string planning_group,
+                SafetyMarginDataConstPtr safety_margin_data,
+                const sco::VarVector& vars)
+  : CollisionEvaluator(planning_scene, planning_group, safety_margin_data), m_vars(vars)
+{
+  // contact_manager_ = env_->getDiscreteContactManager();
+  // contact_manager_->setActiveCollisionObjects(manip_->getLinkNames());
+  // contact_manager_->setContactDistanceThreshold(safety_margin_data_->getMaxSafetyMargin() +
+  //                                              0.04);  // The original implementation added a margin of 0.04;
+}
+
+// So, each pair has a vector of contacts, and this functions puts all of these vectors of all pairs to one big vector called dist_results
+void SingleTimestepCollisionEvaluator::CalcCollisions(const DblVec& x, trajopt::ContactResultVector& dist_results)
+{
+  // tesseract::ContactResultMap contacts;
+  // tesseract::EnvStatePtr state = env_->getState(manip_->getJointNames(), sco::getVec(x, m_vars));
+  // ===> getVec is defined as:
+  // Eigen::VectorXd getVec(const DblVec& x, const VarVector& vars) and
+  // DblVec is vector<double>
+  // VarVec is vector<Var>
+  // Var has VarRep (look in solver_interface) member which has properties like name and index 
+  // getVec extracts the index of VarRep of each Var of a VarVector and finds the element in DblVec with the 
+  // same index and creates the output as VectorXd. 
+  // Finally, getState(joint_names, joint_values)
+  // So, input x of this function, CalcCollisions, is represting the joint values 
+  // joint_values = sco::getVec(x, m_vars);
+
+  // So, I guess I need to set the robot state based on these joint values
+  Eigen::VectorXd joint_values = sco::getVec(x, m_vars);
+  std::vector<double> joint_values_DblVec = std::vector<double>(joint_values.data(), joint_values.data() + joint_values.size());
+  // the following robot state will be a copy not a pointer so it wont make any change in the planning scene
+  moveit::core::RobotState& rob_state = planning_scene_->getCurrentStateNonConst();
+  const moveit::core::JointModelGroup* joint_model_group = rob_state.getJointModelGroup(planning_group_);
+  // rob_state.setToDefaultValues();
+  rob_state.setJointGroupPositions(joint_model_group, joint_values_DblVec);
+  rob_state.update();	 
+  // Here is the point: I created a copy of the robot_state from planning_scene_->getCurrentState()
+  // and updated that copy but that does not change anything in the planning_scene_
+  // as it is a const pointer. const pointer planning scene comes from planning_context and
+  // I can not change that.
+  // planning_scene->setCurrentState(rob_state);
+  std::vector<double> vv;
+  planning_scene_->getCurrentState().copyJointGroupPositions(joint_model_group, vv);
```

Comment:
```suggestion
```
this (including comments) seems superfluous?

Reply:
right, it needs a lot of cleanup :)